server/lease/lessor_test.go

```diff
@@ -208,6 +209,42 @@ func TestLessorRevoke(t *testing.T) {
 	}
 }
 
+func TestLessorGrantAttachRevokeDetach(t *testing.T) {
+	lg := zap.NewNop()
+	_, be := NewTestBackend(t)
+	defer be.Close()
+
+	le := newLessor(lg, be, clusterLatest(), LessorConfig{MinLeaseTTL: minLeaseTTL})
+	defer le.Stop()
+	le.SetRangeDeleter(func() TxnDelete { return newFakeDeleter(be) })
+
+	// grant a lease with long term (100 seconds) to
+	// avoid early termination during the test.
+	l, err := le.Grant(1, 100)
+	if err != nil {
+		t.Fatalf("could not grant lease for 100s ttl (%v)", err)
+	}
+
+	items := []LeaseItem{
+		{"foo"},
+		{"bar"},
+	}
+
+	if err := le.Attach(l.ID, items); err != nil {
+		t.Fatalf("failed to attach items to the lease: %v", err)
+	}
+
```

Comment:
Thanks for the co-authorship, maybe you can add another assertion here that it was indeed put into the itemMap:

> 	assert.Equal(t, 1, len(le.leaseMap))
>	assert.Equal(t, 2, len(le.itemMap))

so that way we can say it was actually deleted, otherwise it might have not been there in the first place :safety_pin:

Reply:
updated